rsocket-core/src/main/java/io/rsocket/core/LeaseHandler.java

```diff
@@ -0,0 +1,8 @@
+package io.rsocket.core;
+
+interface LeaseHandler {
+
+  void handleLease();
+
+  void handleError(Throwable t);
+}
```

Comment:
Maybe rename to `LeasePermitHandler` to make it more clear it handles 1 permit rather than a LEASE frame. The methods could be `handlePermit()` and `handlePermitError(Throwable)` with the latter breaking the overlap with `handleError` from `FrameHandler` which can be difficult to spot otherwise.

Reply:
applied suggestion